fed/mosaic-mapping/src/main/java/org/eclipse/mosaic/fed/mapping/ambassador/MappingAmbassador.java

```diff
@@ -121,46 +123,59 @@ private void handleInteraction(ScenarioTrafficLightRegistration interaction) {
      */
     private void handleInteraction(ScenarioVehicleRegistration scenarioVehicle) throws InternalFederateException {
         if (framework != null) {
-            final CPrototype prototype = framework.getPrototypeByName(scenarioVehicle.getVehicleType().getName());
-            if (prototype == null) {
-                log.debug(
-                        "There is no such prototype \"{}\" configured. No application will be mapped for vehicle \"{}\".",
-                        scenarioVehicle.getVehicleType().getName(),
-                        scenarioVehicle.getId()
-                );
-                return;
-            }
 
-            if (randomNumberGenerator.nextDouble() >= ObjectUtils.defaultIfNull(prototype.weight, 1.0)) {
-                log.debug(
-                        "This scenario vehicle \"{}\" of prototype \"{}\" will not be equipped due to a weight condition of {}.",
-                        scenarioVehicle.getId(),
-                        scenarioVehicle.getVehicleType().getName(),
-                        prototype.weight
-                );
-                return;
+            final List<CPrototype> typeDistribution = framework.getTypeDistributionByName(scenarioVehicle.getVehicleType().getName());
+            if (!typeDistribution.isEmpty()) {
+                final CPrototype selected  = new StochasticSelector<>(typeDistribution, randomNumberGenerator).nextItem();
```

Comment:
I could do this somehow I guess, but I'm afraid that it would blow up the code here and would decrease readability. I'm not sure if this would even improve anything.

Reply:
In the BeST-scenario this would cause ~3 million unnecessary initializations of a `StochasticSelector`, wouldn't it? Also I don't know how the randomness is affected if we draw a type distribution from the same selector or newly initialized ones, although I would assume this doesn't matter.